R/tm_g_ae_oview.R

```diff
@@ -281,83 +285,62 @@ srv_g_ae_oview <- function(id,
           sep = "\n"
         )
       ))
-
-      ANL_UNFILTERED <- datasets$get_data(dataname, filtered = FALSE) # nolint
-      ADSL <- datasets$get_data("ADSL", filtered = TRUE) # nolint
-      ANL <- datasets$get_data(dataname, filtered = TRUE) # nolint
-
-      anl_name <- dataname
-      assign(anl_name, ANL)
-
-      teal.code::chunks_reset(envir = environment())
-
+      ANL <- data[[dataname]]() # nolint
       validate(need(nlevels(ANL[[input$arm_var]]) > 1, "Arm needs to have at least 2 levels"))
```

Comment:
nlevels works only with factors, which means:
1. if `ANL[[arm_var]]` is not a factor the validation message will be printed.
2. (1) means that variable named `arm_var` have to be a factor. There is not other possibility
3. This simplifies our problem with `c(input$arm_trt, input$arm_ref) %in% ANL_UNFILTERED[[input$arm_var]])` because we can define choices in the line 252 as `choices <- levels(ANL[[arm_var]])`